src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -137,6 +141,11 @@ ov::Tensor Bank::eval_and_alloc(const LazyTensor& tensor,
     // Note: this is the non-CPU path!
     const_cast<LazyTensor&>(tensor).detach();
 
+    // Also detach from registered tensors
+    auto it_registered = dbank.registered_tensors.find(tensor);
+    NPUW_ASSERT(it_registered != dbank.registered_tensors.end());
+    const_cast<LazyTensor&>(it_registered->first).detach();
+
     return allocated_tensor;
```

Comment:
This method has a return value but nobody is using it. I'd drop the body of this function completely - it will be easier to do all of this in the parent loop.

Reply:
Done